File: webhooks/csharp/ServeWebhookAgent.cs
using Conductor.AI;
using WebhooksCodelab.Utils;

namespace WebhooksCodelab;

/// <summary>Serve the local database tools used by the post-webhook agent.</summary>
public static class ServeWebhookAgent
{
    /// <summary>Deploy the agent and serve its tool workers until interrupted.</summary>
    public static async Task RunAsync()
    {
        using var cancellation = new CancellationTokenSource();
        Console.CancelKeyPress += (_, eventArgs) =>
        {
            // Let ServeAsync stop its workers cleanly instead of killing the process.
            eventArgs.Cancel = true;
            cancellation.Cancel();
        };

        await using var runtime = new AgentRuntime(Settings.Current.CreateConductorConfiguration());
        Console.WriteLine("Serving the agent's tool workers. Press Ctrl+C to stop.");
        await runtime.ServeAsync(WebhookAgent.Agent, cancellation.Token);
    }
}

File: webhooks/java/src/main/java/io/orkes/codelab/webhooks/ServeWebhookAgent.java
package io.orkes.codelab.webhooks;

import io.orkes.codelab.webhooks.utils.WebhookAgent;

import org.conductoross.conductor.ai.AgentRuntime;

/** Serve the local database tools used by the post-webhook agent. */
public final class ServeWebhookAgent {

    private ServeWebhookAgent() {}

    /** Deploy the agent and serve its tool workers until interrupted. */
    public static int run() {
        try (var runtime = new AgentRuntime(Settings.current().createApiClient())) {
            System.out.println("Serving the agent's tool workers. Press Ctrl+C to stop.");
            // Blocks until the process is stopped; the SDK stops the workers cleanly on Ctrl+C.
            runtime.serve(WebhookAgent.create());
        }
        return 0;
    }
}
